Eugene/MovieSearch/my-app/src/components/MoviePage/MoviePage.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import "./MoviePage.css";
+import IMovie from "../../interface/IMovie/IMovie";
+import MovieDetailHeader from "./MovieDetailHeader/MovieDetailHeader";
+import MovieDetail from "./MovieDetail/MovieDetail";
+import MovieSortPanel from "./MovieSortPanel/MovieSortPanel";
+import MovieList from "../MainPage/movie-list/MovieList";
+import Footer from "../MainPage/footer/Footer";
+
+import { RouteComponentProps } from "react-router-dom";
+
+interface IMoviePageProps extends RouteComponentProps {
+  movies: IMovie[];
+  updateMovies(value: IMovie[]): void;
+}
+
+interface IMoviePageState {
+  movie: IMovie;
+}
+
+class MoviePage extends React.Component<IMoviePageProps, IMoviePageState> {
+  state: IMoviePageState = {
+    movie: {
+      id: 0,
+      title: "",
+      tagline: "",
+      vote_average: 0,
+      vote_count: 0,
+      release_date: "",
+      poster_path: "",
+      overview: "",
+      budget: 0,
+      revenue: 0,
+      genres: [],
+      runtime: 0,
+    },
+  };
+
+  componentDidMount() {
+    this.uploadMovieAndMovieByGenre();
+  }
+
+  componentDidUpdate(prevprops: RouteComponentProps) {
+    if (this.props.location.pathname != prevprops.location.pathname) {
+      this.uploadMovieAndMovieByGenre();
+    }
+  }
+
+  uploadMovieAndMovieByGenre() {
+    fetch(
+      `https://reactjs-cdp.herokuapp.com${this.props.history.location.pathname}`
```

Comment:
If you want to get a param from url better to use **match**. location.pathname might will change and won't be sutable for an api link.